packages/back-end/src/controllers/experiments.ts

```diff
@@ -701,55 +692,48 @@ export async function postExperiment(
     if (hasChanges) {
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
       (changes as any)[key] = data[key];
-      if (keysRequiringWebhook.includes(key)) {
-        requiresWebhook = true;
-      }
     }
   });
 
   // If changing phase start/end dates (from "Configure Analysis" modal)
   if (
-    exp.status !== "draft" &&
+    experiment.status !== "draft" &&
     currentPhase !== undefined &&
-    exp.phases?.[currentPhase] &&
+    experiment.phases?.[currentPhase] &&
     (phaseStartDate || phaseEndDate)
   ) {
-    const phases = [...exp.phases];
+    const phases = [...experiment.phases];
     const phaseClone = { ...phases[currentPhase] };
     phases[Math.floor(currentPhase * 1)] = phaseClone;
 
     if (phaseStartDate) {
       phaseClone.dateStarted = getValidDate(phaseStartDate + ":00Z");
     }
-    if (exp.status === "stopped" && phaseEndDate) {
+    if (experiment.status === "stopped" && phaseEndDate) {
       phaseClone.dateEnded = getValidDate(phaseEndDate + ":00Z");
     }
     changes.phases = phases;
   }
 
-  await updateExperimentById(org.id, exp, changes);
+  const updated = await updateExperimentById(org.id, experiment, changes);
 
   await req.audit({
     event: "experiment.update",
     entity: {
       object: "experiment",
-      id: exp.id,
+      id: experiment.id,
     },
-    details: auditDetailsUpdate(existing, exp),
+    details: auditDetailsUpdate(experiment, updated),
   });
 
   // If there are new tags to add
-  await addTagsDiff(org.id, existing.tags || [], data.tags || []);
+  await addTagsDiff(org.id, updated?.tags || [], data.tags || []);
```

Comment:
This should be `experiment`, not `updated`.  You want to pass both the old tags and the new ones in.  You are passing the new tags twice.